conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -45,7 +47,26 @@ public static <T> void addAll(Collection<T> addTo, Iterable<? extends T> element
         }
     }
 
+    @SuppressWarnings("unchecked")
+    public static <T> void addAllAndCheckNonNull(Collection<T> addTo, Iterable<? extends T> elementsToAdd) {
+        Preconditions.checkNotNull(elementsToAdd, "elementsToAdd cannot be null");
+        // If we know the number of elements we are adding and the addTo Collection is an ArrayList, we can eagerly
+        // resize it to only do one grow() of the array.
+        if (elementsToAdd instanceof Collection) {
+            Collection<T> collectionElementsToAdd = (Collection<T>) elementsToAdd;
+            if (addTo instanceof ArrayList) {
+                ((ArrayList<T>) addTo).ensureCapacity(collectionElementsToAdd.size() + addTo.size());
+            }
+        }
+        for (T element : elementsToAdd) {
+            Preconditions.checkNotNull(element, "elementsToAdd cannot contain null elements");
+            addTo.add(element);
+        }
+    }
+
+    // Prefer to use newList(iterable)
     // explicitly need to return mutable list for generated builders
+    @Deprecated
```

Comment:
Please do not deprecate these, it can cause folks who upgrade conjure-lib prior to upgrading codegen to get a bunch of compiler warnings, making it harder to read the output.
It takes about ~6 months before we can realistically deprecate